code/game/turfs/simulated/walls.dm

```diff
@@ -370,15 +370,34 @@
 		return TRUE
 	return FALSE
 
-/turf/simulated/wall/proc/try_decon(obj/item/I, mob/user, params)
+/turf/simulated/wall/proc/try_decon(obj/item/I, mob/living/user, params)
 	if(istype(I, /obj/item/gun/energy/plasmacutter))
 		to_chat(user, "<span class='notice'>You begin slicing through the outer plating.</span>")
 		playsound(src, I.usesound, 100, 1)
-
 		if(do_after(user, istype(sheet_type, /obj/item/stack/sheet/mineral/diamond) ? 120 * I.toolspeed : 60 * I.toolspeed, target = src))
 			to_chat(user, "<span class='notice'>You remove the outer plating.</span>")
 			dismantle_wall()
-			visible_message("<span class='warning'>[user] slices apart [src]!</span>", "<span class='warning'>You hear metal being sliced apart.</span>")
+			visible_message("<span class='warning'>[user] slices apart [src] with [I]!</span>", "<span class='warning'>You hear metal being sliced apart.</span>")
+			return TRUE
+	if(istype(I, /obj/item/melee/energy))
+		var/obj/item/melee/energy/O = I
+		if(O.can_cut_walls && O.active)
+			to_chat(user, "<span class='notice'>You begin slicing through the outer plating.</span>")
+			playsound(src, 'sound/items/welder.ogg', 100, 1)
+			if(do_after(user, O.toolspeed * 10 SECONDS, target = src))
+				to_chat(user, "<span class='notice'>You remove the outer plating.</span>")
+				dismantle_wall()
+				visible_message("<span class='warning'>[user] slices apart [src] with [O]!</span>", "<span class='warning'>You hear metal being sliced apart.</span>")
```

Comment:
All of these should really be made into their own proc to avoid having the same thing copy+pasted three times.
e.g:
```dm
/turf/simulated/wall/proc/decon_wall(obj/item/I, mob/user, speed, tool_sound)
	to_chat(user, "<span class='notice'>You begin slicing through the outer plating.</span>")
	playsound(src, tool_sound, 100, TRUE)
	if(do_after(user, speed, target = src))
		dismantle_wall()
		user.visible_message("<span class='warning'>[user] slices apart [src] with [I]!</span>", "<span class='notice'>You remove the outer plating.</span>", "<span class='warning'>You hear metal being sliced apart.</span>")
		return TRUE
```

Reply:
Proc made